### PR TITLE
fix(snap): update to use kong deb from bintray

### PR DESCRIPTION
**Pull request by tonyespy**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [NA] Tests for the changes have been added (for bug fixes / features)
- [NA] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## What is the current behavior?

NA

## Issue Number:
NA

## What is the new behavior?
NA

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## New Imports

- [ ] Yes
- [X] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing? No

## Other information
This PR updates the edgexfoundry snap to use the kong debian packages published on bintray vs. building lua, luarocks, kong, and openresty as individual parts (some from source, some downloaded tarballs).

Two other minor changes are included in the PR:

  * filter out man pages included in the snap from the postgresql part
  * strip the date+commit suffix from the snap version

We've verified that kong comes up by default on both amd64 and arm64, and also run the blackbox-tests with security enabled, and found no regressions.

### CODE REVIEW

File: snap/snapcraft.yaml
Comment (anonymouse64) [inline snap/snapcraft.yaml:412]: perhaps also call out this and the postgresql change in the PR description too? and thanks for making those things specific, separate commits

File: snap/snapcraft.yaml
Comment (anonymouse64) [inline snap/snapcraft.yaml:516]: why this change? Won't this add all the go binaries back into the snap?

**Review by anonymouse64 — CHANGES_REQUESTED** [review]
some code in the PR seems leftover from a debugging session?

**Review by lenny-intel — COMMENTED** [review]
LGTM, but will let the Snap experts have final approval.
I did validate version of Kong is correct

**Comment by lenny-intel** [comment]
At Tony, failing Semantic bot due to missing `:` after scope. Should be `fix(snap): update to use kong deb from bintray`

**Comment by tonyespy** [comment]
@anonymouse64 @lenny-intel changes addressed, and re-pushed.

**Review by lenny-intel — DISMISSED** [review]

**Comment by anonymouse64** [comment]
Well I built and tested this PR on amd64 on my focal desktop and everything is groovy :wink: but I haven't been able to build it on arm64 yet to see if it's all good on the pi, I will have to do that on Monday, but I think that is a necessary check that should be done before merging since this is supposed to fix arm64.

**Review by anonymouse64 — CHANGES_REQUESTED** [review]
see previous comment

**Comment by tonyespy** [comment]
@anonymouse64 just a heads-up that @siggiskulason ran the blackbox tests with security enabled against this PR on his RPi4, and although there were a couple of assertion errors thought to be timing related, we're confident that this change works fine on arm64.

**Review by anonymouse64 — APPROVED** [review]
Tested on my arm64 rpi4 running uc18 and kong worked fine in my testing, so lgtm. :tada: 
thanks for figuring this out

**Comment by hutchic** [comment]
I don't have a lot of knowledge about snaps but this seems sensible and who am I to complain about a ~200 line reduction 😆

**Comment by tonyespy** [comment]
recheck

**Review by lenny-intel — APPROVED** [review]

**Pull request merged**
